### ISSUE TITLE
Add colors to error message hints

### ISSUE THREAD

**Issue by Octachron** [ocaml/ocaml#11646]
There is currently 21 hints emitted by various error messages.
For instance,
```html
1 | let f () = raise No_found
                     ^^^^^^^^ 
Error: This variant expression is expected to have type exn
       There is no constructor No_found within type exn
Hint: Did you mean Not_found?
```
The current indentation makes it clear that the `Hint: ...` part is a complement to the error message.
However, currently the `Error:` prefix is strongly emphasized with a red color whereas the `Hint:` prefix
use the standard font setting.

I think it would be useful to print this `Hint:` prefix in a standardized color (blue?) or alternatively with a bold face to makes it even clearer that the `Hint: ...` is a separate part of the error message.

**Comment by bschommer** [comment]
GCC for example uses magenta (with SGR 01;36)

**Comment by ofonime41** [comment]
Hello @Octachron, My name is Christiana Anthony and I am an Outreachy applicant. I have Ocaml setup on my system and will like to work on this issue with your guidance.

**Comment by Octachron** [comment]
You are welcome to try your hand at an implementation.

Color handling is defined in the `Misc.Color` submodule.
The first step is to add a `hint` field to the `styles` record type. Once the new style is defined, you can test your change by using it in one of the `hint` message. However, rather than using this style directly in all error messages, I would recommend to add a function `hint_prefix:Format.formatter -> unit` in the `Location` module that prints directly the `Hint: ` prefix with the right style.

**Comment by ofonime41** [comment]
Alright, I'll get to work rightaway  and update you on my progress. Thank you

**Comment by ofonime41** [comment]
### Hello @Octachron , I added the hint field in the following section  the styles type in the  ml and mli file of the Msc. module. 
### I want to know if I'm on track 
![ocaml1](https://user-images.githubusercontent.com/38950881/196959828-3ac2a5df-1ef4-475a-842f-8a3826023f32.png)
![ocaml2](https://user-images.githubusercontent.com/38950881/196959845-e170f52e-5888-4cd3-bd47-65c1bc611b59.png)

**Comment by Octachron** [comment]
That's look fine. However, it is even better to test your change by trying to add that style to an hint. For instance, you could try to color the two hints that appear in the `Parse.prepare_error` function.

**Comment by Mublin** [comment]
hello, I am an applicant from outreachy, please any issue to work on

**Comment by Oluwabukolab** [comment]
Hello, are you still working on this @ofonime41? If not can I pick it up? @Octachron

**Comment by ofonime41** [comment]
I've open a pull request on this @Octachron. Kindly review. 
https://github.com/ocaml/ocaml/pull/11672

**Comment by gasche** [comment]
This was indeed implemented in #11685, so we can close the present issue. Thanks @ofonime41 for the nice contribution!
